erc777-syntax.k

```diff
@@ -6,40 +6,63 @@ module ERC777-SYNTAX
   imports BOOL-SYNTAX
 
   //
-  // Token API
+  // ERC777Token
   //
 
-  syntax Exp ::= name ( )                              // String
-               | symbol ( )                            // String
-               | totalSupply ( )                       // Number
-               | balanceOf ( Address )                 // Number
-               | granularity ( )                       // Number
-               | defaultOperators ( )                  // Addresses
-               | isOperatorFor ( Address , Address )   // Bool
-
-  syntax Cmd ::= authorizeOperator ( Address )
-               | revokeOperator ( Address )
-               | send ( Address , Number , Data )
-               | operatorSend ( Address , Address , Number , Data , Data )
-               | burn ( Number , Data )
-               | operatorBurn ( Address , Number , Data , Data )
+  syntax Exp ::= AExp | BExp | Set
+
+  syntax Exp ::= name ( )
+               | symbol ( )
+
+  syntax AExp ::= totalSupply ( )
+                | balanceOf ( AExp ) [strict]
+                | granularity ( )
+
+  syntax Exp ::= defaultOperators ( )
+
+  syntax BExp ::= isOperatorFor ( AExp , AExp ) [strict]
+
+  syntax Exp ::=  authorizeOperator ( AExp )                        [strict]
+                | revokeOperator ( AExp )                           [strict]
+                | send ( AExp , AExp , Data )                       [strict]
+                | operatorSend ( AExp , AExp , AExp , Data , Data ) [strict]
+                | burn ( AExp )                                     [strict]
+                | operatorBurn ( AExp , AExp , Data )               [strict]
 
   syntax Event ::= Sent ( Address , Address , Address , Number , Data , Data )
-                 | Minted ( Address , Address , Number , Data )
-                 | Burned ( Address , Address , Number , Data , Data )
+                 | Minted ( Address , Address , Number , Data , Data )
+                 | Burned ( Address , Address , Number , Data )
                  | AuthorizedOperator ( Address , Address )
                  | RevokedOperator ( Address , Address )
 
   //
-  // Token Parameters
+  // ERC777TokensSender
+  //
+
+  syntax Stmt ::= AExp "." "tokensToSend" "(" AExp "," AExp "," AExp "," AExp "," Data "," Data ")" [strict]
+
+  //
+  // ERC777TokensRecipent
+  //
+
+  syntax Stmt ::= AExp "." "tokensReceived" "(" AExp "," AExp "," AExp "," AExp "," Data "," Data ")" [strict]
+
+  //
+  // ERC20 compatibility
   //
 
-  syntax String ::= "#name"   [function]
-                  | "#symbol" [function]
+  // uint256
+  syntax AExp ::= allowance ( AExp , AExp ) [strict]
+                | decimals ( )
 
-  syntax Number ::= "#granularity" [function]
+  // bool
+  syntax BExp ::= approve ( AExp , AExp )             [strict]
+                | transfer ( AExp , AExp )            [strict]
+                | transferFrom ( AExp , AExp , AExp ) [strict]
+                | "revert"
```

Comment:
I'd like to put the prefix `#` for all of the internal symbols for the semantics description purpose.